add a second pass to 'extractTypeLiterals'

0x53A opened the pull request:
extract all type literals which hadn't already been extracted by the first pass.

These won't have the same nice names as in the first pass, because the extractor doesn't have enough context (class name, method name, etc), but will instead be named like "TypeLiteral_%02i"

-----------------------

This is a WIP so I can review the ts2fable-exports.

I am currently writing tests 🙃 

-------------------------

fixes https://github.com/fable-compiler/ts2fable/issues/286

0x53A commented:
ts2fable-exports looks good: https://github.com/fable-compiler/ts2fable-exports/commit/dc9406d#diff-4f589d4

build fail looks unrelated:

![grafik](https://user-images.githubusercontent.com/4236651/53037870-96f87d80-347b-11e9-8715-a51bff0b3ad0.png)

0x53A commented:
I have already prepared some tests and will finish them tomorrow.

What do you think about the general approach?

humhei commented:
> What do you think about the general approach?

OK

humhei commented on test/fsFileTests.fs:
```diff
@@ -316,4 +329,22 @@ describe "transform tests" <| fun _ ->
-        containLineHasWord tsPaths fsPath "Url.URL"
\ No newline at end of file
+        containLineHasWord tsPaths fsPath "Url.URL"
+
```
If a test releated to a github issue
Adding a  link may be better,
So we can track it

0x53A commented:
Test explanation: (I guess I'll add it to the readme)

I added all files in the folder fragments to the project as None, so you can easily navigate them in VS:

**test.fsproj**

```xml
  <ItemGroup>
    <None Include="fragments\**" />
  </ItemGroup>
```

![grafik](https://user-images.githubusercontent.com/4236651/53106848-25c8d100-3534-11e9-9b06-59f83cf913a4.png)


--------

I added two folders: __babylonjs__ and __regressions__.

__babylonjs__ contains excerpts from the babylonjs source code.

__regressions__ contains typescript written by me to reproduce the fixed issues as short and simple as possible.

------

There is always ``*.d.ts`` and ``*.expected.fs``. It compiles the typescript and compares it against the expected source.

I found this easier to author than the other tests, but may create a bit more churn, because the expected files need to be adapter whenever there are _any_ changes to the output.

The easiest way to fix test failures is to run all tests, then copy-paste from the generated fs to expected.fs and review the git diff.
